internal/generate/clusterserviceversion/clusterserviceversion.go

```diff
@@ -160,13 +161,32 @@ func (g *Generator) Generate(cfg *config.Config, opts ...Option) (err error) {
 		return err
 	}
 
+	// Add sdk labels to csv
+	setSDKAnnotations(csv)
+
 	w, err := g.getWriter()
 	if err != nil {
 		return err
 	}
 	return genutil.WriteObject(w, csv)
 }
 
+// setSDKAnnotations adds SDK metric labels to the base if they do not exist. It assumes that
+// if sdk Builder annotation is not present, then all sdk labels are also not populated in csv.
+func setSDKAnnotations(csv *v1alpha1.ClusterServiceVersion) {
+	annotations := csv.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	for key, value := range projutil.MakeOperatorMetricLabels() {
+		if _, exist := annotations[key]; !exist {
```

Comment:
With this existence check, consider the following scenario: I built an operator with v0.18.1 of the SDK, and now I've upgraded to v0.19.0. My CSV still has the annotation `operators.operatorframework.io/builder: operator-sdk-v0.18.1`. The same situation is true for `operators.operatorframework.io/project_layout` when we've bumped plugin versions _or_ when a user migrates their operator from Helm to Ansible or Ansible to Go.

If the generator does not overwrite these annotations, we have broken metrics values. If it does, on-disk files are updated with each SDK version bump (I don't see a problem with this necessarily). I'd prefer to overwrite these values. @joelanford @shawn-hurley FYI.

Reply:
Agreed. Will revert back to overwriting of values.